From: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Subject: mm: make comment consistent in vma_expand()
Date: Thu, 19 Jun 2025 15:32:57 +0100

The VMA merge structure contains a pointer to mm_struct so we don't need
to determine it from an input VMA's vm_mm field.  Update the comment to
reflect this and make it consistent with vma_merge_new_range() and
vma_merge_existing_range().

Link: https://lkml.kernel.org/r/c54f45e3-a6ac-4749-93c0-cc9e3080ee37@lucifer.local
Signed-off-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Jann Horn <jannh@google.com>
Cc: Kees Cook <kees@kernel.org>
Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vma.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/vma.c~mm-vma-use-vmg-target-to-specify-target-vma-for-new-vma-merge-fix
+++ a/mm/vma.c
@@ -1114,7 +1114,7 @@ struct vm_area_struct *vma_merge_new_ran
  * Returns: 0 on success.
  *
  * ASSUMPTIONS:
- * - The caller must hold a WRITE lock on vmg->target->mm->mmap_lock.
+ * - The caller must hold a WRITE lock on the mm_struct->mmap_lock.
  * - The caller must have set @vmg->target and @vmg->next.
  */
 int vma_expand(struct vma_merge_struct *vmg)
_
